jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.c

```diff
@@ -138,10 +138,11 @@ ecma_builtin_string_prototype_object_char_at (ecma_value_t this_arg, /**< this a
 
   /* 4 */
   ecma_string_t *original_string_p = ecma_get_string_from_value (to_string_val);
-  const ecma_length_t len = ecma_string_get_length (original_string_p);
 
   /* 5 */
-  if (index_num < 0 || index_num >= len || !len)
+  if (ecma_string_is_empty (original_string_p)
+      || index_num >= ecma_string_get_length (original_string_p)
+      || index_num < 0)
```

Comment:
There is no performance change on sunspider. I just think it is more readable. I don't mind to change it back, but your previous comment wasn't obvious. :)